packages/block-editor/src/components/block-card/index.js

```diff
@@ -103,6 +106,30 @@ function BlockCard( {
 		( { title, icon, description } = blockType );
 	}
 
+	const { isBlockHidden, hasHiddenParent } = useSelect(
+		( select ) => {
+			if ( ! clientId ) {
+				return { isBlockHidden: false, hasHiddenParent: false };
+			}
+			const { getBlockParents } = select( blockEditorStore );
+			const { isBlockHidden: _isBlockHidden } = unlock(
+				select( blockEditorStore )
+			);
+
+			const blockHidden = _isBlockHidden( clientId );
+			const parents = getBlockParents( clientId );
+			const parentHidden = parents.some( ( parentId ) =>
+				_isBlockHidden( parentId )
+			);
```

Comment:
I expect checking parents' visibility might be needed elsewhere as things progress. Just noting that it might be part of some hook or new selector down the track.